strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/io/RepositoryPath.java

```diff
@@ -234,43 +236,33 @@ public URI toUri()
             return uri;
         }
 
-        RepositoryPath thisPath = this;
-        if (!isAbsolute())
-        {
-            thisPath = thisPath.toAbsolutePath();
-        }
-
         Repository repository = getFileSystem().getRepository();
         Storage storage = repository.getStorage();
-        URI result;
         try
         {
-            result = new URI(StorageFileSystemProvider.STRONGBOX_SCHEME,
-                             null,
-                             "/" + storage.getId() + "/" + repository.getId() + "/",
-                             null);
+            uri = new URI(StorageFileSystemProvider.STRONGBOX_SCHEME,
+                          null,
+                          "/" + storage.getId() + "/" + repository.getId() + "/" +
+                          FilenameUtils.separatorsToUnix(relativize().toString()),
```

Comment:
`relativize().toUri()` should be the same as `FilenameUtils.separatorsToUnix`, I think it will be better to use it

Reply:
`relativize().toUri()` leads to `Method threw 'java.lang.StackOverflowError' exception.` since we are in `toUri` method

Reply:
yeah, I mean `relativize().getTarget().toUri()`

Reply:
Sorry but this is still not the same. See [screenshot](https://ibb.co/895J8qh).

Reply:
this probably should work like this:
```
result = new URI(StorageFileSystemProvider.STRONGBOX_SCHEME,
                             null,
                             "/" + storage.getId() + "/" + repository.getId() + "/",
                             null);
result = result.resolve(relativize().getTarget().toUri());
```

Reply:
ok, let's leave it then